### PR TITLE
Bug/cors

### LINKED ISSUE #103: WebMVC won't allow GraphiQL
Change it to WebFlux so it will run

### PR DESCRIPTION
**Pull request by lmtjanssen**
fixes #103

### CODE REVIEW

**Review by lfb0801 — CHANGES_REQUESTED** [review]
remove everything that has nothing to do with the claimed issue.

File: src/main/java/com/fontys/api/mutations/UserMutation.java [hunk 1]
```diff
@@ -12,5 +12,9 @@
 
     public UserMutation(UserService userService) { this.userService = userService; }
 
-    public User createUser(String UUID) { return userService.createUser(UUID); }
+    public User createUser(String uuid) { return userService.createUser(uuid); }
```
Comment (MindHawk) [inline src/main/java/com/fontys/api/mutations/UserMutation.java]: Not part of issue

File: src/main/java/com/fontys/api/repositories/UserRepository.java [hunk 2]
```diff
@@ -3,7 +3,9 @@
 import com.fontys.api.entities.User;
 import org.springframework.data.jpa.repository.JpaRepository;
 
+import java.util.Optional;
+
 public interface UserRepository extends JpaRepository<User, Integer>
 {
-    User findUserByUuid(String uuid);
+    Optional<User> findByUuid(String UUID);
```
Comment (MindHawk) [inline src/main/java/com/fontys/api/repositories/UserRepository.java]: Not part of issue

File: src/main/java/com/fontys/api/service/UserService.java [hunk 3]
```diff
@@ -14,13 +13,26 @@
...
 
-    public UserService(UserRepository userRepository)
+    UserService(UserRepository userRepository)
     {
         this.userRepository = userRepository;
     }
 
     @Transactional
-    public User createUser(String UUID) { return userRepository.save(new User(UUID)); }
+    public User createUser(String uuid) { return userRepository.save(new User(uuid)); }
```
Comment (MindHawk) [inline src/main/java/com/fontys/api/service/UserService.java]: Not part of issue

File: src/main/java/com/fontys/api/service/UserService.java [hunk 4]
```diff
@@ -29,9 +41,8 @@ public UserService(UserRepository userRepository)
     public Optional<User> getUser(Integer id) { return userRepository.findById(id); }
 
     @Transactional(readOnly = true)
-    public Optional<User> getUserByUUID(String UUID)
+    public Optional<User> getUserByUuid(String uuid)
     {
-        //ToDo: Write option to get user by UUID
-        return Optional.empty();
+        return userRepository.findByUuid(uuid);
```
Comment (MindHawk) [inline src/main/java/com/fontys/api/service/UserService.java]: Not part of issue

File: src/main/resources/graphql/schema.graphqls [hunk 5]
```diff
@@ -10,7 +10,8 @@ type Query {
 }
 
 type Mutation {
-    createUser(UUID:String!):User
+    createUser(uuid:String!):User
+    deleteUser(uuid:String!):String
```
Comment (MindHawk) [inline src/main/resources/graphql/schema.graphqls]: Not part of issue

File: src/test/java/com/fontys/api/service/UserServiceTest.java [hunk 6]
```diff
@@ -20,11 +23,26 @@ void setUp()
...
+    void createUserShouldReturnUser() {
+        User u = new User("UUID");
         when(userRepositoryMock.save(any(User.class))).thenReturn(u);
-        assertEquals(u,userService.createUser("UUID1"));
+        assertEquals(u,userService.createUser("UUID"));
         verify(userRepositoryMock, times(1)).save(u);
     }
+
+    @Test
+    void deleteTeamShouldReturnDeletedString() {
```
Comment (MindHawk) [inline src/test/java/com/fontys/api/service/UserServiceTest.java]: Not part of issue

**Review by MindHawk — CHANGES_REQUESTED** [review]

**Review by MindHawk — APPROVED** [review]
Everything changed is now relevant to the issue.

**Review by larsvandriel — APPROVED** [review]

**Comment by MindHawk** [comment]
Different pull request fixes same issue.

**Pull request closed without merging**